client/blocks/comments/post-comment.jsx

```diff
@@ -196,11 +224,34 @@ class PostComment extends Component {
 	};
 
 	render() {
-		const { commentsTree, commentId, depth, maxDepth } = this.props;
+		const {
+			commentsTree,
+			commentId,
+			depth,
+			enableCaterpillar,
+			maxDepth,
+			post,
+			toShow,
+		} = this.props;
+
 		const comment = get( commentsTree, [ commentId, 'data' ] );
-		const displayType = this.state.showFull
-			? POST_COMMENT_DISPLAY_TYPES.full
-			: this.props.displayType;
+		const isPingbackOrTrackback = comment.type === 'trackback' || comment.type === 'pingback';
+
+		if ( ! comment || ( this.props.hidePingbacksAndTrackbacks && isPingbackOrTrackback ) ) {
+			return null;
+		} else if ( toShow && ! toShow[ commentId ] ) {
+			// this comment should be hidden so just render children
+			return (
+				<div>
```

Comment:
Is it possible that the replies list will be empty here? @jancavan reported empty divs being rendered:

![screenshot_2017-08-30_14 54 18_360](https://user-images.githubusercontent.com/17325/29923955-88ad83be-8e5b-11e7-8130-3a53f42b5af8.png)

Reply:
hmmm.....can do a check for if any children should be rendered and skip the divs that don't apply. I'm worried its more work than rendering empty divs.

I'll get rid of the divs for now but if perf problems arise we'll want to store smarter data structures

Reply:
fixed with https://github.com/Automattic/wp-calypso/pull/17608/commits/a5326ec20a2510ebba93a394b5eb5c144aa893f4